reporter/otlp_reporter.go

```diff
@@ -502,54 +530,40 @@ func (r *OTLPReporter) getResource() *resource.Resource {
 	}
 }
 
-// getProfile returns an OTLP profile containing all collected samples up to this moment.
-func (r *OTLPReporter) getProfile() (profile *profiles.Profile, startTS, endTS uint64) {
-	traceEvents := r.traceEvents.WLock()
-	samples := maps.Clone(*traceEvents)
-	clear(*traceEvents)
-	r.traceEvents.WUnlock(&traceEvents)
-
-	// stringMap is a temporary helper that will build the StringTable.
-	// By specification, the first element should be empty.
-	stringMap := make(map[string]uint32)
-	stringMap[""] = 0
-
-	// funcMap is a temporary helper that will build the Function array
-	// in profile and make sure information is deduplicated.
-	funcMap := make(map[funcInfo]uint64)
-	funcMap[funcInfo{name: "", fileName: ""}] = 0
-
-	// attributeMap is a temporary helper that maps attribute values to
-	// their respective indices.
-	// This is to ensure that AttributeTable does not contain duplicates.
-	attributeMap := make(map[string]uint64)
+// getProfile returns an OTLP profile containing all collected traces up to this moment.
+func (r *OTLPReporter) getProfile(origin int) (profile *profiles.Profile, startTS, endTS uint64) {
```

Comment:
`getProfile()` used to be +200 LOC. I did split it into multiple logical independent functions. So most of the code just shifted and was not changed.